services/cases-api/lambdas/createCase.js

```diff
@@ -5,131 +5,83 @@ import uuid from 'uuid';
 import config from '../../../config';
 
 import * as response from '../../../libs/response';
-import { validateEventBody } from '../../../libs/validateEventBody';
-import { validateKeys } from '../../../libs/validateKeys';
+
 import { decodeToken } from '../../../libs/token';
 import { putItem } from '../../../libs/queries';
 
+import caseValidationSchema from '../helpers/schema';
+import { CASE_STATUS_ONGOING } from '../../../libs/constants';
+import { throwError } from '@helsingborg-stad/npm-api-error-handling';
+
 /**
  * Handler function for creating a case and store in dynamodb
  */
 export async function main(event) {
   const decodedToken = decodeToken(event);
-  const requestBody = JSON.parse(event.body);
 
-  const [validateError, validatedEventBody] = await to(
-    validateEventBody(requestBody, validateCreateCaseRequestBody)
+  const [validationError, validatedEventBody] = await to(
+    validateCreateCaseRequestBody(event.body, caseValidationSchema)
   );
 
-  if (validateError) {
-    return response.failure(validateError);
+  if (validationError) {
+    return response.failure(validationError);
   }
 
-  const TableName = config.cases.tableName;
+  // TODO: check if the passed formId exsists in the form dynamo table.
 
-  const id = uuid.v1();
-  const createdAt = Date.now();
-  const updatedAt = Date.now();
+  const TableName = config.cases.tableName;
 
   const { personalNumber } = decodedToken;
-  const { provider, formId, status, details, answers } = validatedEventBody;
-
-  const PK = `USER#${personalNumber}`;
-  const SK = `USER#${personalNumber}#CASE#${id}`;
+  const { provider, formId } = validatedEventBody;
+  const id = uuid.v4();
```

Comment:
could be put into the caseItem obj to be consistent

Reply:
If you refer to the id, it's used in the SK. So therefore it's defined outside of the caseItem :)